File: apps/whisper-tiny/iOS/whisper/ZeticMLangeWhisper-iOS/ZeticMLangeWhisper-iOS/WhisperEncoder.swift
import Foundation
import ZeticMLange

class WhisperEncoder {
    private var model: ZeticMLangeModel
    
    public static let modelKey = "OpenAI/whisper-tiny-encoder"
    
    init() {
        model = (try? ZeticMLangeModel(personalKey: "YOUR_MLANGE_KEY", name: "YOUR_PROJECT_NAME", version: 1))!
    }
    
    func process(_ audio: [Float32]) -> Data {
        let data = Data(bytes: audio, count: audio.count * MemoryLayout<Float32>.size)
        let tensor = Tensor(data: data, dataType: BuiltinDataType.int8, shape: [data.count])
        return try! model.run(inputs: [tensor])[0].data
    }
    
}

File: apps/whisper-tiny/Android/app/src/main/java/com/zeticai/whisper/WhisperEncoder.kt
package com.zeticai.whisper

import android.content.Context
import com.zeticai.mlange.core.model.ZeticMLangeModel
import com.zeticai.mlange.core.tensor.Tensor
import java.nio.ByteBuffer

class WhisperEncoder(
    context: Context,
    modelKey: String,
    private val model: ZeticMLangeModel = ZeticMLangeModel(context, "YOUR_MLANGE_KEY", modelKey)
) {

    fun process(audioData: FloatArray): ByteBuffer {
        val byteArrayFeatures =  arrayOf(
            Tensor.of(audioData)
        )

        return model.run(byteArrayFeatures)[0].data
    }

    fun close() {
        model.deinit()
    }
}